Graceful Tomcat shutdown

amottier opened the pull request:
Try to gracefully shutdown the Tomcat by sending a shutdown command insted of using System.exit(-1). This development has been done to fix issue with [h2 and quartz shutdown](https://github.com/bonitasoft/bonita-distrib/pull/14#issuecomment-156434786) but is no longer mandatory to fix the issue.

Commit:
Try to gracefully shutdown the Tomcat by sending a shutdown command insted of using System.exit(-1). This development has been done to fix https://github.com/bonitasoft/bonita-distrib/pull/14#issuecomment-156434786 but is no longer mandatory to fix the issue.

Commit:
Fix indentation

apupier commented on src/main/java/org/bonitasoft/console/server/listener/StudioWatchdogListener.java:
```diff
@@ -131,10 +128,38 @@ private static boolean isAlive(final int port, final long timer) throws IOExcept
+            clientSocket.getOutputStream().close();
+            clientSocket.close();
+        } catch (Exception e) {
+            LOGGER.error("Error while trying to shutdown Tomcat from watchdog web application", e);
```
Shouldn't we call System.exit(-1); in catch clause in order to keep the old behavior in case of issue?

apupier commented:
You should update the artifact id version from 6.0.1 to ??? (7.2.0?)

apupier commented on src/main/java/org/bonitasoft/console/server/listener/StudioWatchdogListener.java:
```diff
@@ -19,12 +19,17 @@
 import java.net.InetSocketAddress;
```
Please update copyright BonitaSoft --> Bonitasoft; 2009 -> 2009-2015

apupier commented on src/main/java/org/bonitasoft/console/server/listener/StudioWatchdogListener.java:
```diff
@@ -131,10 +128,38 @@ private static boolean isAlive(final int port, final long timer) throws IOExcept
+            // Connect and send the shutdown command    	    
+            Socket clientSocket = new Socket(address, port);
+            clientSocket.getOutputStream().write(shutdown.getBytes());
+            clientSocket.getOutputStream().flush();
```
I remember of somethign which requires to close all streams even if not used. Not sure it was socket but need to check if we don't need to close also getInputStream

apupier commented:
Can you create the PR to update the version in the studio too? the property watchdog.version in https://github.com/bonitasoft/bonita-studio/blob/master/pom.xml
and link it to this PR

Commit:
Update artifact version to 7.2.0.

Fall back to System.exit(-1) if graceful shutdown fail. Update copyright
and author

apupier commented:
Can you updat ethe message of last commit   "Update artifact version to 7.0.2." although it is 7.2.0

Commit:
Add error handling on socket and output stream

apupier commented on src/main/java/org/bonitasoft/console/server/listener/StudioWatchdogListener.java:
```diff
@@ -131,10 +128,38 @@ private static boolean isAlive(final int port, final long timer) throws IOExcept
+            // Get the server object
+            Server server = (Server) mBeanServer.getAttribute(name, "managedResource");
+
+            // Get adress and port on which we can send the shutdown command
```
typo adress -> address

amottier commented:
I update copyright and artifact version (7.2.0, error in commit message reporting 7.0.2 I cannot edit it).
I also add a fallback to System.exit(-1); as recommended.
And finally I add some error handling on Socket and OutputStream.

Commit:
Force shutdown command charset to UTF-8. Fix comment typo
